addons/mass_mailing/models/mail_thread.py

```diff
@@ -50,3 +51,15 @@ def message_post_with_template(self, template_id, **kwargs):
             default_mass_mailing_id=False,
         )
         return super(MailThread, no_massmail).message_post_with_template(template_id, **kwargs)
+
+    @api.multi
+    def message_receive_bounce(self, email, partner, mail_id=None):
```

Comment:
I think this one deserve a nice comment because I bet we will soon forget why we did this :) .